File: SkyEditor.SaveEditor/MysteryDungeon/Explorers/IExplorersActivePokemon.vb
Namespace MysteryDungeon.Explorers
    Public Interface IExplorersActivePokemon

        ''' <summary>
        ''' The index of the Pokemon in storage as stored in the save file.
        ''' </summary>
        ''' <returns></returns>
        Property RosterNumber As Integer

        Property Level As Byte
        Property ID As Integer
        Property IsFemale As Boolean
        Property MetAt As Integer
        Property MetFloor As Integer
        Property IQ As Integer
        Property HP1 As Integer 'Todo: rename
        Property HP2 As Integer 'Todo: rename
        Property Attack As Byte
        Property Defense As Byte
        Property SpAttack As Byte
        Property SpDefense As Byte
        Property Exp As Integer
        Property Name As String
        ReadOnly Property PokemonNames As Dictionary(Of Integer, String)
        ReadOnly Property LocationNames As Dictionary(Of Integer, String)

        Property Attack1 As ExplorersActiveAttack
        Property Attack2 As ExplorersActiveAttack
        Property Attack3 As ExplorersActiveAttack
        Property Attack4 As ExplorersActiveAttack


    End Interface
End Namespace

File: SkyEditor.SaveEditor/MysteryDungeon/Explorers/IExplorersActivePokemon.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace SkyEditor.SaveEditor.MysteryDungeon.Explorers
{
    public interface IExplorersActivePokemon
    {
        int RosterNumber { get; set; }
        int Level { get; set; }
        ExplorersPokemonId ID { get; set; }
        int MetAt { get; set; }
        int MetFloor { get; set; }
        int IQ { get; set; }
        int CurrentHP { get; set; }
        int MaxHP { get; set; }
        int Attack { get; set; }
        int Defense { get; set; }
        int SpAttack { get; set; }
        int SpDefense { get; set; }
        int Exp { get; set; }
        string Name { get; set; }

        ExplorersActiveAttack Attack1 { get; set; }
        ExplorersActiveAttack Attack2 { get; set; }
        ExplorersActiveAttack Attack3 { get; set; }
        ExplorersActiveAttack Attack4 { get; set; }
    }
}
